### PR TITLE
Adding ThM (modified throttle send to engine) to EFIS log

### PR DESCRIPTION
**Pull request by loki077**
- ThM added to help understand the engine logs better
 - Updated the field names to keep the total length 64-characters long.

### CODE REVIEW

File: libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp [hunk 1]
```diff
@@ -205,6 +205,7 @@ void AP_EFI_Serial_Hirth::send_request()
 bool AP_EFI_Serial_Hirth::send_target_values(uint16_t thr)
 {
     uint8_t computed_checksum = 0;
+    throttle = 0;
```
Comment (peterbarker) [inline libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp:208]: This is a little too generic a name for a member variable IMO.

Perhaps
```suggestion
    throttle_to_hirth = 0;
```

File: libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp [hunk 2]
```diff
@@ -369,23 +370,24 @@ void AP_EFI_Serial_Hirth::log_status(void)
...
-    // @Field: ETSS: Status of Engine temperature sensor
-    // @Field: ATSS: Status of Air temperature sensor
-    // @Field: APSS: Status of Air pressure sensor
+    // @Field: ES1: Status of EGT1 excess temperature error
+    // @Field: ES2: Status of EGT2 excess temperature error
+    // @Field: CS1: Status of CHT1 excess temperature error
+    // @Field: CS2: Status of CHT2 excess temperature error
+    // @Field: ETS: Status of Engine temperature sensor
+    // @Field: ATS: Status of Air temperature sensor
+    // @Field: APS: Status of Air pressure sensor
```
Comment (peterbarker) [inline libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp:379]: Many of these are booleans.  You could save columns and logging bandwidth by logging them into a bitmask.

Many analysis tools do have reasonable support for bitmasks now.

**Review by peterbarker — CHANGES_REQUESTED** [review]
Commit message needs fixing.

File: libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp (hunk 2 above)
Comment (loki077) [reply libraries/AP_EFI/AP_EFI_Serial_Hirth.cpp]: I have change the implementation and grouped all bool into 2 groups EET( Error Excess Temperature) FLAGS ( Sensor Status) I can also group them into one but thought to keep separate @peterbarker let me know your opinion on that. 
Also, will WriteStreaming be able to use FieldBitmaskEnum?

image
image

**Comment by loki077** [comment]
> Commit message needs fixing.

I will fix the commit message once the code is all good.

**Comment by loki077** [comment]
@Pradeep-Carbonix @robertlong13 FYI

File: libraries/AP_EFI/AP_EFI_Serial_Hirth.h [hunk 3]
```diff
@@ -31,6 +31,28 @@ class AP_EFI_Serial_Hirth: public AP_EFI_Backend {
 
     void update() override;
 
+    enum class Error_Excess_Temp_Bitfield : uint16_t {
```
Comment (IamPete1) [inline libraries/AP_EFI/AP_EFI_Serial_Hirth.h:34]: These ones that are only really used for log documentation you can define right in the `log_status` function in the cpp. That stops it getting used elsewhere my mistake.

Comment (loki077) [reply libraries/AP_EFI/AP_EFI_Serial_Hirth.h:34]: @IamPete1 what i mention above is that ok?

Comment (tridge) [reply libraries/AP_EFI/AP_EFI_Serial_Hirth.h:34]: seems reasonable to me

**Review by tridge — APPROVED** [review]

**Pull request merged**